Shopify/ruby-lsp
# Centralize request handling logic in server.rb

### Motivation

Currently, the steps to add a new request are:

1. Define new provider in `handler.rb`'s `respond_with_capabilities` method
1. Define the related `respond_with_*` method in `handler.rb`
1. Define the supported request type (which usually just calls the above method) in `cli.rb`
1. Add a new request class and call it in the `respond_with_*` method in `handler.rb`

IMO the jumping between files is confusing. And I think in group pairings we also mention that these steps can be shorten if we don't define the `respond_with_*` methods.

After this PR, the flow will become

1. Define new provider in `server.rb`'s `initialize` request
1. Define the supported request type in `server.rb`
1. Add a new request class and call it in the above block `server.rb`

It has less steps and the contributor can focus on only `server.rb`.

### Implementation

- Move all `respond_with_*` methods into `Handler#on` blocks.
- Move the request definitions to `server.rb` and remove `cli.rb`.

### Automated Tests

Don't need any new tests but all old tests should pass.

### Manual Tests

1. Restart RubyLSP
2. Check there's no error in ruby-lsp's extension output
3. Check features like document highlight works as expected
